### PR TITLE
improve build scripts

### PR DESCRIPTION
**Pull request by iosmanthus**
Signed-off-by: iosmanthus <myosmanthustree@gmail.com>

### What problem does this PR solve? 
* add shell.nix for Nixer
* refine `srcipts/proto.sh`

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+(
+  pkgs.buildFHSUserEnv {
+    name = "client-java-shell";
+    targetPkgs = pkgs: with pkgs;[ maven openjdk11 git ];
```
Comment (marsishandsome) [inline shell.nix:6]: currently, client-java is compiled with java8.

Comment (iosmanthus) [reply shell.nix:6]: OK

**Comment by iosmanthus** [comment]
@marsishandsome PTAL

**Review by marsishandsome — APPROVED** [review]
LGTM

**Review by tisonkun — APPROVED** [review]

**Pull request merged**